api/src/main/java/org/openmrs/module/kenyaemr/advice/SyncMCHAppointmentsWithBahmniModule.java

```diff
@@ -0,0 +1,145 @@
+
+        if (method.getName().equals("saveEncounter")) { // handles both create and edit
+            Encounter enc = (Encounter) args[0];
+            Person parent = personService.getPerson(enc.getPatient().getPersonId());
+            Appointment mchAppointment = appointmentsService.getAppointmentByUuid(enc.getUuid());
+
+            List<Obs> obs = obsService.getObservations(
+                    Arrays.asList(personService.getPerson(enc.getPatient().getPersonId())),
+                    Arrays.asList(enc),
+                    Arrays.asList(
+                            conceptService.getConceptByUuid(NEXT_APPOINTMENT_DATE_CONCEPT_UUID),
+                            conceptService.getConceptByUuid(CWC_NEXT_VISIT_DATE_CONCEPT_UUID)
+                    ),
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    false
+            );
+
+            for (Obs o : obs) { // Loop through the obs and compose Appointment object for Bahmni
+                AppointmentServiceDefinition appointmentServiceDefinition = new AppointmentServiceDefinition();
+                SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+
+                if(enc.getVoided() == true && mchAppointment != null && enc.getForm() != null) {
+                    // void appointment
+
+                    mchAppointment.setVoided(true);
+                    mchAppointment.setDateVoided(new Date());
+                    mchAppointment.setVoidedBy(Context.getAuthenticatedUser());
+                    Appointment app = appointmentsService.validateAndSave(mchAppointment);
+
+                } else if(mchAppointment != null && enc.getForm() != null && enc.getForm() != null && (enc.getForm().getUuid().equals(MchMetadata._Form.MCHMS_DELIVERY) || enc.getForm().getUuid().equals(MchMetadata._Form.MCHMS_POSTNATAL_VISIT) || enc.getForm().getUuid().equals(MchMetadata._Form.MCHMS_ANTENATAL_VISIT) || enc.getForm().getUuid().equals(MchMetadata._Form.MCHCS_FOLLOW_UP))) {
+                    //edit appointment
+
+                    Date nextApptStartDateTime = DateUtil.convertToDate(dateFormat.format(o.getValueDatetime()).concat("T07:00:00.0Z"), DateUtil.DateFormatType.UTC);
+                    Date nextApptEndDateTime = DateUtil.convertToDate(dateFormat.format(o.getValueDatetime()).concat("T20:00:00.0Z"), DateUtil.DateFormatType.UTC);
+                    mchAppointment.setStartDateTime(nextApptStartDateTime);
+                    mchAppointment.setEndDateTime(nextApptEndDateTime);
+                    if(enc.getVoided() == true) {
+                        mchAppointment.setVoided(true);
+                        mchAppointment.setDateVoided(new Date());
+                        mchAppointment.setVoidedBy(Context.getAuthenticatedUser());
+                    }
```

Comment:
Not sure why we need this here. Please help me understand.

Reply:
I will do the necessary